src/eventbuffer/InMemoryJSONEventBuffer.ts

```diff
@@ -519,15 +482,21 @@ export default class InMemoryJSONEventBuffer implements EventBuffer<EventData>,
         },
         body: data,
       });
-      return response;
+      if (response.ok || !InMemoryJSONEventBuffer.SENDING_FAILURE_CODES.has(response.status)) {
+        return response;
+      } else {
+        const newRetryCount = retryCount + 1;
+        await this.wait(this.getBackoffWaitTime(newRetryCount));
+        return this.send(data, newRetryCount);
```

Comment:
I suggest to avoid recursion and implement retry logic as a loop.

Reply:
Agree :)

Reply:
Cool, thanks for the explanation offline. I have updated retrying logic to use loop over recursion.